File: Thread/demo/nativeTid/app/src/main/java/com/example/nativetid/MainActivity.java
package com.example.nativetid;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.example.nativetid.databinding.ActivityMainBinding;

import java.lang.reflect.Field;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'native-lib' library on application startup.
    static {
        System.loadLibrary("native-lib");
    }

    private ActivityMainBinding binding;
    private final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
        try {
            init();
        } catch (IllegalAccessException e) {
            e.printStackTrace();
        }
    }

    private void init() throws IllegalAccessException {
        Thread thread = new Thread("demo_thread"){
            @Override
            public void run() {
                try {
                    sleep(10 * 1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                Log.d(TAG, "->>>Thread对象的run方法被执行了");
            }
        };
        //线程启动
        thread.start();
        long peer = getNativePeer(thread);
        Log.d(TAG, "->>>tid = " + getTid(peer));

    }

    public static final long getNativePeer(Thread t)throws IllegalAccessException{
        try {
            Field nativePeerField = Thread.class.getDeclaredField("nativePeer");
            nativePeerField.setAccessible(true);
            Long nativePeer = (Long) nativePeerField.get(t);
            return nativePeer;
        } catch (NoSuchFieldException e) {
            throw new IllegalAccessException("failed to get nativePeer value");
        } catch (IllegalAccessException e) {
            throw e;
        }
    }
    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
    public static native int getTid(long nativePeer);
}

File: MyRecyclerView/app/src/main/java/com/example/myrecyclerview/MainActivity.kt
package com.example.myrecyclerview

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button


/**
 * as版本4.2.1
 * gradle版本 6.7.1
 * compileSdkVersion 30
 */
class MainActivity : AppCompatActivity() , View.OnClickListener{
    companion object {
        private const val TAG = "MainActivity"
    }

    private var bt1: Button? = null
    private var bt2: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView()
    }

    private fun initView() {
        bt1 = findViewById(R.id.btSimpleRecycerView)
        bt2 = findViewById(R.id.btWifisettings)
        bt1?.setOnClickListener(this)
        bt2?.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when( v?.id)
        {
            R.id.btSimpleRecycerView ->
            {
                Log.d(TAG, "->onclick|btSimpleRecycerView")
                startActivity(Intent(this, SimpleRecyclerView::class.java))
            }

            R.id.btWifisettings ->
            {
                Log.d(TAG, "->onclick|btWifisettings")
                startActivity(Intent(this, WifiSettings::class.java))
            }

        }
    }




}